pmd-java/src/main/resources/category/java/bestpractices.xml

```diff
@@ -1444,6 +1444,62 @@ public class Foo {
         </example>
     </rule>
 
+    <rule name="UseTryWithResources"
+          language="java"
+          minimumLanguageVersion="1.7"
+          since="6.11.0"
+          message="Consider use the try-with-resources statement instead of explicitly closing the resource"
+          class="net.sourceforge.pmd.lang.rule.XPathRule"
+          externalInfoUrl="${pmd.website.baseurl}/pmd_rules_java_bestpractices.html#usetrywithresources">
+        <description>
+Java 7 introduced the try-with-resources statement. This statement ensures that each resource is closed at the end
+of the statement. It avoids the need of explicitly closing the resources in a finally block. Additionally exceptions
+are better handled: If an exception occurred both in the `try` block and `finally` block, then the exception from
+the try block was suppressed. With the `try`-with-resources statement, the exception thrown from the try-block is
+preserved.
+        </description>
+        <priority>3</priority>
+        <properties>
+            <property name="closeMethods" type="List[String]" delimiter="," description="Method names in finally block, which trigger this rule" value="close,closeQuietly"/>
+            <property name="version" value="2.0" />
+            <property name="xpath">
+                <value>
+<![CDATA[
+//TryStatement[not(ResourceSpecification)][FinallyStatement//Name[
```

Comment:
so, having:

```java
Reader r1 = new FileReader(file);
try (Reader r2 = new FileReader(otherFile)) {
} finally {
  r1.close();
}
```

would produce no violations? I'm unsure `[not(ResourceSpecification)]` is actually useful here...

Reply:
You're right - that would produce a false negative. I'm removing the check for ResourceSpecifications. Half-converted try-with-resources statements should also be considered.